CommonTools/RecoAlgos/plugins/JetDeltaRValueMapProducer.cc

```diff
@@ -166,7 +166,7 @@ class JetDeltaRValueMapProducer : public edm::EDProducer {
   const std::vector<std::string>                   valueLabels_;
   const bool                                       lazyParser_;
   bool                                             multiValue_;
-  std::map<std::string, StringObjectFunction<T> >  evaluationMap_;
+  std::map<std::string, std::unique_ptr<const StringObjectFunction<T> > >  evaluationMap_;
```

Comment:
I walked my way through the code and indeed StringObjectFunction is not thead safe. See ExpressionLazyVar with its mutable member. However, StringObjectFunction does appear to be thread-friendly (i.e. different instances do not interfere with one another).